auth/service.go

```diff
@@ -10,18 +10,35 @@ import (
 	"github.com/haleyrc/cheevos/lib/hash"
 )
 
+type IRepository interface {
```

Comment:
I don't really know what to name these. This is one of those "no-no" scenarios where I'm defining an interface _and_ and implementation of that interface in the same package, but for good reason. The Hungarian-esque notation thing just smells though. Unfortunately, the simple alternative also makes build errors pretty useless.